app/javascript/submission_handling/shared.js

```diff
@@ -49,4 +49,9 @@ export const scrollToTop = () => {
         "left": 0,
         "behavior": 'smooth'
     });
+};
+
+export const clearBadges = () => {
+    const badges = document.getElementsByClassName('badge-success');
+    while (badges.length > 0) badges[0].remove();
```

Comment:
Overall comment: This approach seems fine though a bit tricky, like a race maybe between clearing old badges and clearing new ones by mistake in the case where new badges appear very quickly. 

What about adding a new listener to ajax:before? And in that listener you can trigger a badge clear. See https://guides.rubyonrails.org/working_with_javascript_in_rails.html#rails-ujs-event-handlers

Aside: stylistically I would prefer curly brackets as `js-lint` [is asking for](https://eslint.org/docs/2.0.0/rules/curly) - this is kinda hard to read

Reply:
@cdmo would you mind doing a another quick review for the changes I added?

Reply:
You didn't use a new listener for ajax:before... ?

Reply:
I moved the clearBadges calls to existing `ajax:success`listeners for example https://github.com/psu-libraries/myaccount/pull/268/files#diff-c144f09703f116bdd78fd8aee6f84d69R40-R42

Reply:
Ok, and I think that's better, but you didn't use a new listener for ajax:before and my comment was just wondering why. It seemed to me like a handy way to make sure, for sure, that the badges are all cleared out before any of the ajax requests, spinning and value updating occurred. What you have here is probably fine though 👍

Reply:
clearing before spinning and etc. makes sense, let me change to add ajax:before

Reply:
Ok thanks. It's probably fine how you have it. I just get worried about weird things happening in JavaScript land when it comes to order of DOM manipulation.